endpoint/route/route.go

```diff
@@ -123,15 +125,6 @@ func New(ctx context.Context, c client.Client, logger logr.Logger,
 		return nil, err
 	}
 
-	healthy, err := r.IsHealthy(ctx, c)
```

Comment:
By pulling out IsHealthy(), are we changing the semantics of the Endpoint interface?

Reply:
@JohnStrunk No, this is in the `New()` function. It's not an interface at the moment. The problem is that when you call `New()` it creates an object and also checks if it's healthy, but in reality, it takes some time for resource to become healthy. So `New()` can return an error when the objects are created but are not healthy yet...So IMO, `New()` should simply create objects and return creation errors. `IsHealthy()` should be called explicitely by the caller.

Reply:
I also believe we should introduce builder to better handle creation logic